apps/visibleplanets/visibleplanets.star

```diff
@@ -0,0 +1,399 @@
+                                    child = render.Text(row1, color = "#ffff00", font = "5x8"),
+                                ),
+                                render.Marquee(
+                                    width = 64,
+                                    child = render.Text(row2, color = "#ffff00", font = "5x8"),
+                                ),
+                            ],
+                        ),
+                    ],
+                ),
+            ],
+        ),
+    )
+
+def get_magnitude_description(magnitude):
+    """ Gets a description of the visibility based on the magnitude
+
+    Args:
+        magnitude: scale used to determine relative brightness of objects
+    Returns:
+        Description of that brightness
+    """
+    description = ""
+    if magnitude < 0:
+        description = "easily visible"
+    elif magnitude < 4:
+        description = "visible to naked eye"
+    elif magnitude < 5:
+        description = "visible w/ binoculars"
+    elif magnitude < 6:
+        description = "near limit of naked eye"
+    elif magnitude < 10:
+        description = "visible with small telescope"
+    elif magnitude < 30:
+        description = "near limit of Hubble telescope"
+    else:
+        description = "you might need the James Webb Space telescope"
+
+    return description
+
+def get_cardinal_position_from_degrees(bearing):
+    """ Returns the cardinal position for a given bearing
+
+    Args:
+        bearing: in degrees
+    Returns:
+        The Cardinal position (N, NW, NE, S, SW, SE, E, W)
+    """
+
+    if bearing < 0:
+        bearing = 360 + bearing
+
+    # have bearning in degrees, not convert to cardinal point
+    compass_brackets = ["North", "NNE", "NE", "ENE", "East", "ESE", "SE", "SSE", "South", "SSW", "SW", "WSW", "West", "WNW", "NW", "NNW", "North"]
+    display_cardinal_point = compass_brackets[int(math.round(bearing // 22.5))]
+    return display_cardinal_point
+
+def get_body_position(body, location, check_time):
+    """ Gets the JSon Data from astronomyapi
+
+    Args:
+        body: Which celestial body we want info on
+        location: Location of the observer
+        check_time: Time to get the data for the given planet
+    Returns:
+        JSon Data
+    """
+
+    date_code = "%s-%s-%s" % (two_character_time_date_part(check_time.year), two_character_time_date_part(check_time.month), two_character_time_date_part(check_time.day))
+    tomorrow_date_code = check_time + time.parse_duration("24h")
+    tomorrow_date_code = "%s-%s-%s" % (two_character_time_date_part(tomorrow_date_code.year), two_character_time_date_part(tomorrow_date_code.month), two_character_time_date_part(tomorrow_date_code.day))
+    time_code = "%s:%s:%s" % (two_character_time_date_part(check_time.hour), two_character_time_date_part(check_time.minute), two_character_time_date_part(check_time.second))
+
+    params = {
+        "body": body,
+        "longitude": location["lng"],
```

Comment:
We need lat/lng to be truncated when passed out to external APIs. This to avoid leaking users' location data. In this case, I think 1 decimal point will work just find. Did a few experiments and got pretty consistent results that way. As a bonus, it also means users who're located nearby can rely on each other's cached data so we get fewer API calls overall.

Reply:
I originally kept the precision to get the right sunrise/sunset, but I refactored such that I have the precision for sunset/sunrise times, but use the rounded coordinates for api calls and the cache name.